include/RAJA/util/IndexLayout.hpp

```diff
@@ -0,0 +1,166 @@
+/*!
+ ******************************************************************************
+ *
+ * \file
+ *
+ * \brief   RAJA header file defining the IndexLayout class and IndexList classes.
+ *
+ ******************************************************************************
+ */
+
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2016-23, Lawrence Livermore National Security, LLC
+// and RAJA project contributors. See the RAJA/LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#ifndef RAJA_INDEXLAYOUT_HPP
+#define RAJA_INDEXLAYOUT_HPP
+
+#include "RAJA/util/Layout.hpp"
+
+namespace RAJA 
+{
+
+/*!
+* DirectIndex struct contains call operator that returns the same index that was input
+*
+*/
+template<typename IdxLin = Index_type>
+struct DirectIndex {
+
+  IdxLin RAJA_INLINE RAJA_HOST_DEVICE constexpr operator()(const IdxLin idx) const
+  {
+    return idx;
+  }
+
+};
+
+/*!
+* IndexList struct stores a pointer to an array containing the index list.
+* Its call operator returns the entry at the input location (idx) of its index list.
+* 
+*/
+template<typename IdxLin = Index_type>
+struct IndexList {
+
+  IdxLin* index_list{nullptr};
+
+  IdxLin RAJA_INLINE RAJA_HOST_DEVICE constexpr operator()(const IdxLin idx) const
+  {
+    return index_list[idx];
+  }
+
+};
+
+/*!
+* ConditionalIndexList struct stores a pointer to an array containing the index list.
+* Its call operator returns the same index that was input if the index list is a nullptr, 
+* or otherwise returns the entry at the input location (idx) of its index list.
+* 
+*/
+template<typename IdxLin = Index_type>
+struct ConditionalIndexList {
+
+  IdxLin* index_list{nullptr};  
+
+  IdxLin RAJA_INLINE RAJA_HOST_DEVICE constexpr operator()(const IdxLin idx) const
+  {
+    if (index_list) return index_list[idx];
+    else return idx;
```

Comment:
I think our best practices/style guide says to always use curly-braces.
```suggestion
    if (index_list) {
      return index_list[idx];
    } else {
      return idx;
    }
```

Reply:
Fixed.